osu.Game/Beatmaps/Drawables/BeatmapGroup.cs

```diff
@@ -1,42 +1,46 @@
-//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
-//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
-
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using osu.Framework;
-using osu.Framework.Graphics;
-using osu.Framework.Graphics.Transformations;
-using osu.Game.Database;
-
-namespace osu.Game.Beatmaps.Drawables
-{
-    class BeatmapGroup : IStateful<BeatmapGroupState>
-    {
-        public BeatmapPanel SelectedPanel;
-
-        /// <summary>
-        /// Fires when one of our difficulties was selected. Will fire on first expand.
-        /// </summary>
-        public Action<BeatmapGroup, BeatmapInfo> SelectionChanged;
-
+//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Transformations;
+using osu.Game.Database;
+
+namespace osu.Game.Beatmaps.Drawables
+{
+    class BeatmapGroup : IStateful<BeatmapGroupState>
+    {
+        public BeatmapPanel SelectedPanel;
+
+        /// <summary>
+        /// Fires when one of our difficulties was selected. Will fire on first expand.
+        /// </summary>
+        public Action<BeatmapGroup, BeatmapInfo> SelectionChanged;
+
         /// <summary>
         /// Fires when one of our difficulties is clicked when already selected. Should start playing the map.
-        /// </summary>
-        public Action<BeatmapInfo> StartRequested;
-
-        public BeatmapSetHeader Header;
-
+        /// </summary>
+        public Action<BeatmapInfo> StartRequested;
+
+        public BeatmapSetHeader Header;
+
         private BeatmapGroupState state;
+        
+        public List<BeatmapPanel> BeatmapPanels;
+
+        public BeatmapSetInfo BeatmapSet;
 
-        public List<BeatmapPanel> BeatmapPanels;
-
-        public BeatmapGroupState State
-        {
-            get { return state; }
-            set
-            {
-                state = value;
+        public bool Hidden;

```

Comment:
not sure i like this being a public bool with no actual logic behind it. changing this value should either not be allowed or should actually perform an action. consider making it private if it's just storing a local state.